cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/NamedConfiguration.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.batch.dataset;
+
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.JobContext;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Utility class for prefixing a set of configurations with a name and also for extracting the prefixed parameters.
+ */
+public class NamedConfiguration {
```

Comment:
Seems like the functionality is pretty generic. Consider adding them to `ConfigurationUtil` in common.